helper/helper-common/src/main/java/com/linkedin/avroutil1/compatibility/FieldBuilder.java

```diff
@@ -21,7 +21,7 @@
 
   FieldBuilder setDoc(String doc);
 
-  FieldBuilder setDefault(Object defaultValue);
+  FieldBuilder setDefault(Object defaultValue, Schema schema);
```

Comment:
the field has a schema. accepting a potentially different schema as arg here opens the door to various bugs where field schema and default value schema are completely unrelated.

maybe instead we assert that the field schema has already been set at this point, and derive the default value schema from there?

Reply:
Done. Also added better comments, more tests and updated the PR description. PTAL.